type_inference_visitor.py

```diff
@@ -0,0 +1,126 @@
+from astroid.transforms import TransformVisitor
+import astroid
+import warnings
+from typing import Tuple, List, Dict
+
+
+class TypeVisitor(TransformVisitor):
+
+    def _visit(self, node):
```

Comment:
Sure, I suppose that means all the behaviour is specified using `register_transform`